src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -202,6 +200,11 @@ private static Expression CreateArgument(ParameterInfo parameter, FactoryContext
 
             if (parameterCustomAttributes.OfType<IFromRouteMetadata>().FirstOrDefault() is { } routeAttribute)
             {
+                if (factoryContext.RouteParameters is { } routeParams && !routeParams.Contains(parameter.Name))
+                {
+                    throw new InvalidOperationException($"{parameter.Name} is not a route paramter.");
```

Comment:
This is worth a look. I added some tests for this scenario as well. TL;DR If you specify `[FromRoute]` and but didn't specify a route parameter in the template, this will throw.

Reply:
Not identical, but Blazor does this today and it helps alleviate the class of issues. We can always change it to be less restrictive if we users run in to in the future (particularly given that we have an option type for configure this).